src/front/glsl/variables.rs

```diff
@@ -331,9 +331,23 @@ impl Program<'_> {
             return Ok(ctx.add_expression(Expression::Constant(handle), body));
         }
 
-        let class = match storage {
-            StorageQualifier::StorageClass(class) => class,
-            _ => StorageClass::Private,
+        let (class, storage_access) = match self.module.types[ty].inner {
+            TypeInner::Image { class, .. } => (
+                StorageClass::Handle,
+                if let ImageClass::Storage(_) = class {
+                    StorageAccess::all()
```

Comment:
this should depend on `readonly`/`writeonly`/`readwrite` qualifiers

Reply:
I know, I just haven't gotten around to implementing them, I'll add a TODO